charts/gitops-server/templates/role.yaml

```diff
@@ -35,4 +35,7 @@ rules:
   - apiGroups: [ "" ]
     resources: [ "namespaces" ]
     verbs: [ "get", "list" ]
+  - apiGroups: [ "apps" ]
```

Comment:
Why are you granting more permissions to the service account role? Shouldn't the call to query deployments be using the user's permissions via impersonation?

As far as I can tell this is being added to fix tests and we shouldn't be adding permissions to the chart to fix those.

Reply:
While I was implementing the `Flux Runtime` page, I realize that most users won't have access to listing deployments in the flux namespace, causing them to always see a red box with an error message, so I added that permission and used the service account credentials to query for deployments in that specific page only, everything else uses impersonated clients.

Reply:
That feels like how you'd want it to operate though? It might be that we want to surpress that specific error message for users we think just can't see that namespace but if I've not granted a user permissions to see deployments in a specific namespace I don't want them to be able to see those deployments

Reply:
I take your point @luizbafilho on the tab being useless if the user lacks the permissions to view deployments in the flux-system namespace, however I think we're better off leaning more towards consistent use of impersonation for now. This is something we could speak to users/customers about and change later based on their feedback. As an alternative, could this be configurable? Namely to document changes that an administrative user (one who deploys Weave GitOps) could make to enable the use of the service account with increased permissions?

Reply:
Sounds good, that's probably doable, we can investigate it in a follow up issue.

Reply:
@SamLR access removed, using user's credentials instead.